ocw/lib/gce.py

```diff
@@ -3,31 +3,39 @@
 import googleapiclient.discovery
 from google.oauth2 import service_account
 from dateutil.parser import parse
+from .decorators import filterService
+from googleapiclient.errors import HttpError
 import re
 
 
 class GCE(Provider):
     __instances = dict()
 
-    def __new__(cls, vault_namespace):
-        if vault_namespace not in GCE.__instances:
-            GCE.__instances[vault_namespace] = self = object.__new__(cls)
-            self.__credentials = GCECredential(vault_namespace)
-            self.__compute_client = None
-            self.__project = None
+    def __new__(cls, vault_namespace, *args, **kwargs):
+        if vault_namespace in GCE.__instances:
+            return GCE.__instances[vault_namespace]
+        GCE.__instances[vault_namespace] = object.__new__(cls)
         return GCE.__instances[vault_namespace]
 
+    def __init__(self, namespace):
+        self.__credentials = GCECredential(namespace)
+        self.__compute_client = None
+        self.__project = None
+        self.credentials = None
+        super().__init__(namespace)
```

Comment:
this is always best practice and also here it is quite necessary AFAIK to call parent's init as very first thing to do

Reply:
*this change is ready for review*